packages/compose-tunnel-agent/src/api-server/index.ts

```diff
@@ -0,0 +1,104 @@
+import fs from 'node:fs'
+import url from 'node:url'
+import { WebSocketServer } from 'ws'
+import { Logger } from 'pino'
+import Dockerode from 'dockerode'
+import { SshState } from '../ssh'
+import { NotFoundError, respond, respondAccordingToAccept, respondJson, tryHandler, tryUpgradeHandler, tryWsHandler } from './http-server-helpers'
+import { DockerFilterClient } from '../docker'
+import { findHandler as findWsHandler, handlers as wsHandlers } from './ws'
+import { ContainerNotFoundError, MissingContainerIdError } from './errors'
+
+// const pathRe = /^\/(?<resourceType>[^/]+)(\/(?<resourceId>[^/]+)(?<action>\/[^/]+)?)?$/
+
+const createApiServerHandlers = ({
+  log,
+  currentSshState,
+  machineStatus,
+  envMetadata,
+  composeModelPath,
+  dockerFilter,
+  docker,
+}: {
+  log: Logger
+  currentSshState: () => Promise<SshState>
+  machineStatus?: () => Promise<{ data: Buffer; contentType: string }>
+  envMetadata?: Record<string, unknown>
+  composeModelPath: string
+  dockerFilter: DockerFilterClient
+  docker: Dockerode
+}) => {
+  const handler = tryHandler({ log }, async (req, res) => {
+    const { pathname: path } = url.parse(req.url || '')
+
+    if (req.method === 'GET' && path === '/healthz') {
```

Comment:
Not specifically for this PR, but teed to add support for CORS, so these apis can be fetched also from client.
@eliyabar currently add it locally for testing

Reply:
done